### PR TITLE
Simplify protobuf linking

### PR DESCRIPTION
**Pull request by cyyever**
### Description

Move logic of Protobuf_USE_STATIC_LIBS to find_package(protobuf) because it's a CMake module variable but not a protobuf build option, don't print it in the summary and remove it from README.md. The PROTOBUF_USE_DLLS should not be set at all to avoid subtle linking errors. It also align MSVC_STATIC_RUNTIME with protobuf's. 
```
 cmake_minimum_required(VERSION 3.11)
```
is also removed since the top line has 
```
cmake_minimum_required(VERSION 3.14)
```
Finally, some typos of README.md are fixed.

### CODE REVIEW

File: cmake/summary.cmake [hunk 1]
```diff
@@ -21,7 +21,6 @@ function (onnx_print_configuration_summary)
   message(STATUS "  ONNX NAMESPACE                    : ${ONNX_NAMESPACE}")
   message(STATUS "  ONNX_USE_LITE_PROTO               : ${ONNX_USE_LITE_PROTO}")
   message(STATUS "  USE_PROTOBUF_SHARED_LIBS          : ${ONNX_USE_PROTOBUF_SHARED_LIBS}")
-  message(STATUS "  Protobuf_USE_STATIC_LIBS          : ${Protobuf_USE_STATIC_LIBS}")
```
Comment (xadupre) [inline cmake/summary.cmake:24]: Why remove this line?

File: CMakeLists.txt [hunk 2]
```diff
@@ -16,7 +16,7 @@ cmake_policy(SET CMP0074 NEW)
 
 # Project
 project(onnx C CXX)
-option(ONNX_USE_PROTOBUF_SHARED_LIBS "Build ONNX using protobuf shared library. Sets PROTOBUF_USE_DLLS CMAKE Flag and Protobuf_USE_STATIC_LIBS. " OFF)
+option(ONNX_USE_PROTOBUF_SHARED_LIBS "Build ONNX using protobuf shared library." OFF)
```
Comment (xadupre) [inline CMakeLists.txt:19]: More information is usually better. Why remove it?

File: cmake/summary.cmake (hunk 1 above)
Comment (cyyever) [reply cmake/summary.cmake:24]: Now Protobuf_USE_STATIC_LIBS is set by ONNX_USE_PROTOBUF_SHARED_LIBS  and has the opposite value of ONNX_USE_PROTOBUF_SHARED_LIBS. It should not be set directly by the user, so it's not printed.

File: CMakeLists.txt (hunk 2 above)
Comment (cyyever) [reply CMakeLists.txt:19]: These are CMake variables for finding Protobuf, they are package finding details and should not be exposed and set by the user. For example, if the user chooses to build a static ONNX lib and set the PROTOBUF_USE_DLLS macro, there may be linking error.

**Review by justinchuby — APPROVED** [review]
lgtm, thanks!

**Pull request merged**